feature/tabsflow/search/src/main/java/com/epmedu/animeal/tabs/search/presentation/ui/SearchScreenUi.kt

```diff
@@ -145,10 +151,34 @@ private fun ScreenScaffold(
             }
         )
     }
-    if (state.feedState.feedingConfirmationState
-        == FeedingConfirmationState.FeedingStarted
-    ) {
-        navigator.navigateTo(TabsRoute.Home.name)
+    OnFeedingConfirmationStateChange(
+        navigator,
+        state.feedState.feedingConfirmationState
+    )
+}
+
+@Composable
+fun OnFeedingConfirmationStateChange(
+    navigator: Navigator,
+    feedingConfirmationState: FeedingConfirmationState,
+) {
+    val feedingWarningDialogHasShown = rememberSaveable { mutableStateOf(false) }
+    when (feedingConfirmationState) {
+        FeedingStarted -> { navigator.navigateTo(TabsRoute.Home.name) }
+        is Dismissed -> {
+            val showWarningDialog =
+                feedingConfirmationState.reason == REQUEST_CANCELLED
+            if (showWarningDialog && !feedingWarningDialogHasShown.value) {
```

Comment:
Here is a small bug, we don't return feedingWarningDialogHasShown.value back to false, so user won't see error dialogs next times. Plus, feedingConfirmationState.reason still contains the error message, which, probably, should have been nullified after user confirmed he has seen an error dialog.
But ignore those, let's just redo it using separate feedingConfirmationState altogether as I suggested in another comment

Reply:
Sure, with the new confirmation state this shouldn't be an issue. Will return to Dismissed state when user click it.